### ISSUE TITLE
Should @Header, @Headers and @HeaderMap support target type?

### ISSUE THREAD

**Issue by VictorAlbertos** [square/retrofit#1866]
Currently these 3 `annotations` limit its `target` to params. It would be nice to extend their funcionality to support `target` `type` too. Frequently, I need to repeat some headers, e.g, api key, method by method. 

I know I can use an `interceptor` for that matter; but, still, I prefer to keep my interface decorate with as much annotations as possible; because this metadata serves very well to ilustrate the REST api specifications.

I could try to submit a PR, in case this would be deemed appropriate.

**Comment by swankjesse** [comment]
As soon as this exists we need precedence rules. That might break the benefits of this. (DRY vs. bigger abstractions).

**Comment by VictorAlbertos** [comment]
One simple approach could be throwing an `exception` when a header is present on both param and type. That way there would be not need to set precedence rules.

**Comment by JakeWharton** [comment]
`@Header` and `@HeaderMap` require a parameter value so I'm assuming you are only referring to `@Headers` which is the only one of the three whose data exists completely statically? We've avoided adding any type-level annotations so far and I'm not eager to really start. How many endpoints are you defining?

**Comment by VictorAlbertos** [comment]
You are right, `@Headers` would be the only one elegible to be supported as type-level annotation (I've updated the tittle of the issue to reflect it). 

About the number of endpoints which I define, well, it depends on the application. The current one has about 40.  

I just can use an `interceptor`. But I thought that a type-level `@Headers` could reflect better this kind of requirement.
